weatherApp/App/Data/WeatherRepository.swift

```diff
@@ -19,81 +20,89 @@ class WeatherRepository {
+            return Observable.of(currentWeatherListCoreData)
+            
         default:
-            weatherService.fetchSeveralCurrentWeather(id: id) { [weak self] currentWeatherList in
-                guard
-                    let self = self,
-                    !currentWeatherList.isEmpty
-                else {
-                    completion([])
-                    return
-                }
-                for currentWeather in currentWeatherList {
-                    self.coreDataService.createCurrentWeatherFrom(currentWeather: currentWeather)
-                }
-                self.coreDataService.saveChanges()
-                
-                let currentWeatherListCoreData = self.coreDataService.fetchCurrentWeather()
-                completion(currentWeatherListCoreData)
-            }
+            return weatherService.fetchSeveralCurrentWeather(id: id)
+                .do(onNext: { [weak self] multipleCurrentWeather in
+                    guard let self = self else { return }
+                    
+                    let currentWeatherList = multipleCurrentWeather.list
+                    for currentWeather in currentWeatherList {
+                        self.coreDataService.createCurrentWeatherFrom(currentWeather: currentWeather)
+                    }
+                    self.coreDataService.saveChanges()
+                })
+                .flatMap({ multipleCurrentWeather -> Observable<[CurrentWeatherCoreData]> in
+                    let currentWeatherListCoreData = self.coreDataService.fetchCurrentWeather()
+                    return Observable.of(currentWeatherListCoreData)
+                })
         }
     }
     
-    func fetchForcastWeather(coord: Coordinates, completion: @escaping ((ForecastedWeatherCoreData?) -> Void)) {
+    func fetchForcastWeather(coord: Coordinates) -> Observable<ForecastedWeatherCoreData> {
         switch reachability.connection {
         case .unavailable:
-            let currentWeatherListCoreData = coreDataService.fetchForecastWeather(coord: coord)
-            completion(currentWeatherListCoreData)
-        default:
-            weatherService.fetchForcastWeather(coord: coord) { [weak self] forecastedWeather in
-                guard
-                    let self = self,
-                    let forecastedWeather = forecastedWeather
-                else {
-                    completion(nil)
-                    return
-                }
-                
-                self.coreDataService.createForecastedWeatherFrom(forecastedWeather: forecastedWeather)
-                self.coreDataService.saveChanges()
-                
-                let currentWeatherListCoreData = self.coreDataService.fetchForecastWeather(coord: coord)
-                completion(currentWeatherListCoreData)
+            guard let forecastedWeather = coreDataService.fetchForecastWeather(coord: coord) else {
+                return Observable.of()
```

Comment:
You can set ForecastedWeatherCoreData as optional. And then you can return .just(nil) here. Note You don't need to write Observable.just(nil). You can write this as:
```
return .just(nil)
```